src/Symfony/Component/HttpFoundation/AcceptHeader.php

```diff
@@ -147,4 +180,145 @@ private function sort(): void
             $this->sorted = true;
         }
     }
+
+    /**
+     * Sorts candidates by specificity, quality, parameter count, and original order.
+     *
+     * @param AcceptHeaderItem[] $candidates
+     */
+    private function sortCandidates(array &$candidates, AcceptHeaderItem $queryItem): void
+    {
+        usort($candidates, function (AcceptHeaderItem $a, AcceptHeaderItem $b) use ($queryItem) {
+            $specificityA = $this->getSpecificity($a, $queryItem);
+            $specificityB = $this->getSpecificity($b, $queryItem);
+            if ($specificityA !== $specificityB) {
+                return $specificityB <=> $specificityA;
+            }
+            if ($a->getQuality() !== $b->getQuality()) {
+                return $b->getQuality() <=> $a->getQuality();
+            }
+            $parameterCountA = \count($this->getMediaParams($a));
+            $parameterCountB = \count($this->getMediaParams($b));
+            if ($parameterCountA !== $parameterCountB) {
+                return $parameterCountB <=> $parameterCountA;
+            }
+
+            return $a->getIndex() <=> $b->getIndex();
+        });
+    }
+
+    /**
+     * Builds a canonical string representation of attributes.
+     *
+     * @param array<string, mixed> $attributes
+     */
+    private function buildCanonicalParams(array $attributes): string
+    {
+        $normalizedAttributes = [];
+        foreach ($attributes as $name => $value) {
+            $lowerName = strtolower($name);
+            if ('q' === $lowerName) {
+                continue;
+            }
+            $normalizedAttributes[$lowerName] = $value;
+        }
```

Comment:
Then we could simplify to: 
```suggestion
        unset($attributes['q]);
```